ImageDetection/ImageDetection/blob_detector.py

```diff
@@ -0,0 +1,132 @@
+    The following code and descriptions are adapted from OpenCV:
+    """
+    def __init__(self, image_path, target_size_range_in_pixels):
+        """
+        Read image.
+        """
+        self.image_path = image_path
+        self.target_size_range_in_pixels = target_size_range_in_pixels
+        self.raw_image = Image.open(image_path)
+        self.minimum_area = target_size_range_in_pixels[0]
+        self.maximum_area = (target_size_range_in_pixels[1])**1.5
+
+    def detect_blobs(self):
+        """
+        The algorithm of SimpleBlobDetector follows the following steps:
+
+        Thresholding: Convert the source images to several binary images by
+                      thresholding the source image with thresholds starting at
+                      minThreshold. These thresholds are incremented by
+                      thresholdStep until maxThreshold. So the first threshold
+                      is minThreshold, the second is
+                      minThreshold + thresholdStep, the third is
+                      minThreshold + 2 x thresholdStep, and so on.
+
+        Grouping: In each binary image, connected white pixels are grouped
+                  together. Let us call these binary blobs.
+
+        Merging: The centers of the binary blobs in the binary images are
+                 computed, and blobs located closer than minDistBetweenBlobs are
+                 merged.
+
+        Center & Radius Calculation: The centers and radii of the new merged
+                                     blobs are computed and returned.
+        """
+
+        """
+        Set up the detector with default parameters.
+        """
+        params = cv2.SimpleBlobDetector_Params()
+
+        """
+        Set up Thresholds. Values are changed in ImageDetectionSettings.
+        """
+        params.minThreshold = ImageDetectionSettings.MINIMUM_THRESHOLD
+        params.maxThreshold = ImageDetectionSettings.MAXIMUM_THRESHOLD
+
+        """
+        Set up filters. Values are changed in ImageDetectionSettings.
+        """
+        params.filterByArea = ImageDetectionSettings.FILTER_BY_AREA_ON
+        params.minArea = self.minimum_area
+        params.maxArea = self.maximum_area
+
+        params.filterByCircularity = ImageDetectionSettings.FILTER_BY_CIRCULARITY_ON
+        params.minCircularity = ImageDetectionSettings.MINIMUM_CIRCULARITY
+        params.maxCircularity = ImageDetectionSettings.MAXIMUM_CIRCULARITY
+
+        params.filterByConvexity = ImageDetectionSettings.FILTER_BY_CONVEXITY_ON
+        params.minConvexity = ImageDetectionSettings.MINIMUM_CONVEXITY
+        params.maxConvexity = ImageDetectionSettings.MAXIMUM_CONVEXITY
+
+        params.filterByInertia = ImageDetectionSettings.FILTER_BY_INERTIA_ON
+        params.minInertiaRatio = ImageDetectionSettings.MINIMUM_INERTIA_RATIO
+        params.maxInertiaRatio = ImageDetectionSettings.MAXIMUM_INERTIA_RATIO
+
+        """
+        Create a detector with the parameters.
+        """
```

Comment:
Single-line comments should begin with a `#` and not use the `"""` comment begin/close.